src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -233,6 +236,32 @@ public static bool TryParse(string value, bool allowFloating, out VersionRange v
                 }
             }
 
+            // Illogical version range detection
+            if (minVersion != null && maxVersion != null
+                && minVersion >= maxVersion
+                && !(minVersion == maxVersion && isMinInclusive && isMaxInclusive))  // Exclude (9.0.0,9.0.0) since it's used as empty version range.
+            {
+                if (partsLength == 1)
+                {
+                    var onePartValidation = isMinInclusive ^ isMaxInclusive; // (1.0.0] and [1.0.0) are invalid
+                    onePartValidation |= !isMinInclusive && !isMaxInclusive; ; // or (1.0.0) is invalid
+                    if (onePartValidation)
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    var manyPartsValidation = minVersion > maxVersion // If a > b then (a, b) range is invalid. But below 2 cases tests are a = b (equals).
+                                              | isMinInclusive // [1.0.0, 1.0.0) is invalid
```

Comment:
This check is pretty much equivalent to the above one. 

I'd take a look at [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws).